packages/openapi3-parser/test/unit/parser/oas/parseSchemaObject-test.js

```diff
@@ -60,114 +70,206 @@ describe('Schema Object', () => {
-        type: 'null',
+    describe('when type is an array', () => {
+      beforeEach(() => {
+        context = new Context(namespace);
+        context.openapiVersion = { major: 3, minor: 1 };
       });
-      const parseResult = parse(context, schema);
 
-      expect(parseResult).to.contain.warning(
-        "'Schema Object' 'type' must be either boolean, object, array, number, string, integer"
-      );
-    });
+      it('warns when using array on OpenAPI 3.0', () => {
+        context.openapiVersion = { major: 3, minor: 0 };
 
-    it('returns a null structure for null type on OpenAPI 3.1', () => {
-      context.openapiVersion = { major: 3, minor: 1 };
-      const schema = new namespace.elements.Object({
-        type: 'null',
+        const schema = new namespace.elements.Object({
+          type: [],
+        });
+        const parseResult = parse(context, schema);
+
+        expect(parseResult).to.contain.warning(
+          "'Schema Object' 'type' is not a string"
+        );
       });
-      const parseResult = parse(context, schema);
 
-      expect(parseResult.length).to.equal(1);
-      expect(parseResult.get(0)).to.be.instanceof(namespace.elements.DataStructure);
-      expect(parseResult).to.not.contain.annotations;
+      it('warns when array items is empty', () => {
+        const schema = new namespace.elements.Object({
+          type: [],
+        });
+        const parseResult = parse(context, schema);
 
-      const element = parseResult.get(0).content;
-      expect(element).to.be.instanceof(namespace.elements.Null);
+        expect(parseResult).to.contain.warning(
+          "'Schema Object' 'type' array must contain at least one type"
+        );
+      });
+
+      it('warns when array items are not a string', () => {
+        const schema = new namespace.elements.Object({
+          type: [true],
+        });
+        const parseResult = parse(context, schema);
+
+        expect(parseResult).to.contain.warning(
+          "'Schema Object' 'type' array value is not a string"
+        );
+      });
+
+      it('warns when array items are not valid types', () => {
+        const schema = new namespace.elements.Object({
+          type: ['invalid'],
+        });
+        const parseResult = parse(context, schema);
+
+        expect(parseResult).to.contain.warning(
+          "'Schema Object' 'type' array must only contain values: boolean, object, array, number, string, integer, null"
+        );
+      });
+
+      it('warns when array items are duplicated', () => {
+        const schema = new namespace.elements.Object({
+          type: ['string', 'string'],
+        });
+        const parseResult = parse(context, schema);
+
+        expect(parseResult).to.contain.warning(
+          "'Schema Object' 'type' array must contain unique items, string is already present"
+        );
+      });
+
+      it('warns when array items contain more than 1 entry', () => {
+        // FIXME support more than one entry :)
```

Comment:
This is the one test that is more temporary as it should go away once we do support multiple types.